src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx

```diff
@@ -458,27 +437,57 @@ export default function TerraformEditView(
         }}
       >
         <TFMonaco
-          name="spec.configuration.spec.backendConfig.spec.content"
+          name={`spec.${fieldPath}.spec.backendConfig.spec.content`}
           formik={formik as FormikProps<unknown>}
           expressions={expressions}
           title={getString('cd.backEndConfig')}
         />
       </MultiTypeFieldSelector>
-      {getMultiTypeFromValue(formik.values.spec?.configuration?.spec?.backendConfig?.spec?.content) ===
+      {getMultiTypeFromValue(get(formik.values.spec, `${fieldPath}.spec.backendConfig.spec.content`)) ===
         MultiTypeInputType.RUNTIME && (
         <ConfigureOptions
-          value={formik.values.spec?.configuration?.spec?.backendConfig?.spec?.content as string}
+          value={get(formik.values.spec, `${fieldPath}.spec.backendConfig.spec.content`) as string}
           type="String"
-          variableName="spec.configuration.spec.backendConfig.spec.content"
+          variableName={`spec.${fieldPath}.spec.backendConfig.spec.content`}
           showRequiredField={false}
           showDefaultField={false}
-          onChange={value => formik.setFieldValue('spec.configuration.spec.backendConfig.spec.content', value)}
+          onChange={value => formik.setFieldValue(`spec.${fieldPath}.spec.backendConfig.spec.content`, value)}
           isReadonly={readonly}
         />
       )}
     </div>
   )
 
+  const [enableCloudCli, setEnableCloudCli] = React.useState<boolean | undefined>(undefined)
+
+  useEffect(() => {
+    setEnableCloudCli(prevEnableCloudCli => {
+      if (isUndefined(prevEnableCloudCli)) {
+        return (
+          !isEmpty((formikRefValues?.current?.values as TerraformData)?.spec?.cloudCliConfiguration) &&
+          !isUndefined((formikRefValues?.current?.values as TerraformData)?.spec?.cloudCliConfiguration)
+        )
+      }
+    })
```

Comment:
This is fine but can you also return the previous state when the if condition fails?

```suggestion
    setEnableCloudCli(prevEnableCloudCli => {
      if (isUndefined(prevEnableCloudCli)) {
        return (
          !isEmpty((formikRefValues?.current?.values as TerraformData)?.spec?.cloudCliConfiguration) &&
          !isUndefined((formikRefValues?.current?.values as TerraformData)?.spec?.cloudCliConfiguration)
        )
      }
      return prevEnableCloudCli
    })
```